File: python-vectorembeddingweaknesses/secure8.py
from langchain_community.vectorstores import Chroma
from langchain_openai import OpenAIEmbeddings
from fastapi import FastAPI, HTTPException
from pydantic import BaseModel
import re

app = FastAPI()
embeddings = OpenAIEmbeddings()
vectorstore = Chroma(collection_name="docs", embedding_function=embeddings)

INJECTION_PATTERNS = re.compile(
    r'(ignore previous|you are now|system prompt|forget your|new instructions)',
    re.IGNORECASE,
)

class IngestRequest(BaseModel):
    text: str

@app.post("/ingest")
def ingest(req: IngestRequest):
    if INJECTION_PATTERNS.search(req.text):
        raise HTTPException(status_code=400, detail="Content rejected")
    vectorstore.add_texts([req.text])
    return {"status": "ok"}

File: typescript-vectorembeddingweaknesses/secure8.ts
import express, { Request, Response } from "express";
import { Chroma } from "@langchain/community/vectorstores/chroma";
import { OpenAIEmbeddings } from "@langchain/openai";

const app = express();
app.use(express.json());

const embeddings = new OpenAIEmbeddings();
const vectorStore = new Chroma(embeddings, { collectionName: "docs" });

app.post("/search", async (req: Request, res: Response) => {
  const { query }: { query: string } = req.body;
  const results = await vectorStore.similaritySearch(query, 5);
  res.json({ results: results.map((r) => r.pageContent) });
});

app.listen(3000);
